File: android/app/src/main/kotlin/samples/flutter/devbattery/battery_connection_2/EventChannelRegistry.kt
package samples.flutter.devbattery.battery_connection_2

import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel

class EventChannelRegistry private constructor() {
    companion object {
        val shared = EventChannelRegistry()
        private const val TIMEZONE_CHANNEL = "timeHandlerEvent"
    }

    fun setupEventChannels(binaryMessenger: BinaryMessenger, context: Context) {
        EventChannel(binaryMessenger, TIMEZONE_CHANNEL).setStreamHandler(
            TimezoneListener(context)
        )
    }
}

File: ios/Runner/EventChannelRegistry.swift
import Flutter
import Foundation

class EventChannelRegistry {
  static let shared = EventChannelRegistry()
  
  private init() {}
  
  func setupEventChannels(with controller: FlutterViewController) {
    let timezoneChannel = FlutterEventChannel(
      name: "timeHandlerEvent",
      binaryMessenger: controller.binaryMessenger
    )
    timezoneChannel.setStreamHandler(TimezoneListener())
  }
}
